src/handlers.js

```diff
@@ -263,6 +327,80 @@ async function statusHandler(message) {
   }
 }
 
+/**
+ * Post updates to GitHub, when the status of a task changes. Uses Checks API
+ **/
+async function statusHandler(message) {
+  let {taskGroupId, state, runs, taskId} = message.payload.status;
+  let {runId} = message.payload;
+  let {reasonResolved} = runs[runId];
+
+  let {organization, repository, sha, eventId, eventType, installationId} = await this.context.Builds.load({
+    taskGroupId,
+  });
+
+  let debug = Debug(`${debugPrefix}:${eventId}`);
+  debug(`Handling state change for task ${taskId} in group ${taskGroupId}`);
+
+  let taskState = {
+    status: 'completed',
+    conclusion: CONCLUSIONS[reasonResolved || state],
+    completed_at: new Date().toISOString(),
+  };
+
+  // true means we'll get null if the record doesn't exist
+  let checkRun = await this.context.CheckRuns.load({taskGroupId, taskId}, true);
+
+  // Authenticating as installation.
+  try {
+    debug('Authenticating as installation in status handler...');
+    var instGithub = await this.context.github.getInstallationGithub(installationId);
+    debug('Authorized as installation in status handler');
+  } catch (e) {
+    debug(`Error authenticating as installation in status handler! Error: ${e}`);
+    throw e;
+  }
+
+  debug(
+    `Attempting to update status of the checkrun for ${organization}/${repository}@${sha} (${taskState.conclusion})`
+  );
+  try {
+    if (checkRun) {
+      await instGithub.checks.update({
+        ...taskState,
+        owner: organization,
+        repo: repository,
+        check_run_id: checkRun.checkRunId, // TODO: a helpful output
+      });
+    } else {
+      const checkRun = await instGithub.checks.create({
+        owner: organization,
+        repo: repository,
+        name: `Task ${taskId}`, // <-- this needs to be unique 😩
```

Comment:
What if we fetch the task definition and use `task.metadata.name` here?

Reply:
I had thought about it, and I don't like that option because I don't want to rely on users for having unique `task.metadata.name`. It's like "Hey, I can't solve this problem, so please be inconvenienced", and I doubt that's a realistic expectation anyways. People just copy-paste metadata generally. What will happen is this: then they will see one checkrun for one task, while their YAML has 5. They will file a bug in tc-gh. (and that will be fair!)

I plan to think about this some more. It's not to ask for help, more like a `TODO`. However, I don't want this to be a blocker for the first pass either.

Reply:
I mean, even if I implement a check for `task.metadata.name` uniqueness in `tc-yml`... I wonder if that would make creating tc-yml more complicated and obnoxious... 
> "Hey, I can't solve this problem, so please be inconvenienced"